trio/_highlevel_generic.py

```diff
@@ -72,34 +77,34 @@ class StapledStream(HalfCloseableStream, metaclass=Final):
 
     """
 
-    send_stream = attr.ib()
-    receive_stream = attr.ib()
+    send_stream: SendStream = attr.ib()
+    receive_stream: ReceiveStream = attr.ib()
 
-    async def send_all(self, data):
+    async def send_all(self, data: bytes | bytearray | memoryview) -> None:
         """Calls ``self.send_stream.send_all``."""
         return await self.send_stream.send_all(data)
 
-    async def wait_send_all_might_not_block(self):
+    async def wait_send_all_might_not_block(self) -> None:
         """Calls ``self.send_stream.wait_send_all_might_not_block``."""
         return await self.send_stream.wait_send_all_might_not_block()
 
-    async def send_eof(self):
+    async def send_eof(self) -> None:
         """Shuts down the send side of the stream.
 
         If ``self.send_stream.send_eof`` exists, then calls it. Otherwise,
         calls ``self.send_stream.aclose()``.
 
         """
         if hasattr(self.send_stream, "send_eof"):
-            return await self.send_stream.send_eof()
+            return await self.send_stream.send_eof()  # type: ignore
         else:
             return await self.send_stream.aclose()
 
-    async def receive_some(self, max_bytes=None):
+    async def receive_some(self, max_bytes: int | None = None) -> bytes | bytearray:
```

Comment:
`trio-typing` says *only* bytes. But given that it didn't include `bytearray` above, I guessed that should maybe be here as well?
And what about `memoryview`?

Reply:
No, I think in this case we accept sending `bytearray` from the caller, but only provide `bytes` to the caller (_ala_ Postel's Principle).  That's also consistent with the `_highlevel_socket` implementation.